test/e2e/chaos/pod.go

```diff
@@ -184,6 +186,86 @@ var _ = ginkgo.Describe("[chaos-mesh] Basic", func() {
 		framework.ExpectNoError(err, "Pod kill chaos perform failed")
 		cancel()
 	})
+
+	ginkgo.It("PauseChaos", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		var err error
+		for _, i := range []int{1, 2, 3} {
+			name := fmt.Sprintf("nginx-%d", i)
+			bpod := fixture.NewCommonNginxPod(name, ns)
+			_, err = kubeCli.CoreV1().Pods(ns).Create(bpod)
+			framework.ExpectNoError(err, fmt.Sprint("create pod", name, "error"))
+			err = waitPodRunning(name, ns, kubeCli)
+			framework.ExpectNoError(err, fmt.Sprint("wait pod", name, "error"))
+		}
+
+		podKillChaos := &v1alpha1.PodChaos{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "nginx-kill",
+				Namespace: ns,
+			},
+			Spec: v1alpha1.PodChaosSpec{
+				Selector: v1alpha1.SelectorSpec{
+					Namespaces:     []string{ns},
+					LabelSelectors: map[string]string{"app": "nginx"},
+				},
+				Action: v1alpha1.PodKillAction,
+				Mode:   v1alpha1.OnePodMode,
+				Scheduler: &v1alpha1.SchedulerSpec{
+					Cron: "@every 10s",
+				},
+			},
+		}
+		err = cli.Create(ctx, podKillChaos)
+		framework.ExpectNoError(err, "create pod chaos error")
+
+		var mergePatch []byte
+		mergePatch, _ = json.Marshal(map[string]interface{}{
+			"spec": map[string]interface{}{
+				"paused": true,
+			},
+		})
+
+		err = cli.Patch(ctx, podKillChaos, client.ConstantPatch(types.MergePatchType, mergePatch))
+		framework.ExpectNoError(err, "patch pause pod chaos error")
+
+		chaosKey := types.NamespacedName{
+			Namespace: ns,
+			Name:      "nginx-kill",
+		}
+
+		err = wait.Poll(5*time.Second, 5*time.Minute, func() (done bool, err error) {
+			chaos := &v1alpha1.PodChaos{}
+			err = cli.Get(ctx, chaosKey, chaos)
+			if chaos.Status.Experiment.Phase == chaosmeshv1alpha1.ExperimentPhasePaused {
+				return true, nil
+			}
+			return false, err
+		})
+		framework.ExpectNoError(err, "Check paused chaos failed")
```

Comment:
Almost lgtm, but the e2e case should be revised. I think we could record the pods uid after we create the nginx Deployments. Then, we apply the pod-kill chaos and checking wether is there any pod get killed by comparing the pods uid. Next, we patch the pause into pod-kill chaos. After that, we record the pods uid again and expect a timeout error from checking the pods whether haven't been deleted for 5 mins.

Reply:
@Yisaer PTAL at the new e2e test